crates/cairo-addons/src/vm/runner.rs

```diff
@@ -748,104 +743,55 @@ except Exception as e:
     let execution_resources = cairo_runner.get_execution_resources().unwrap();
     tracing::info!("Execution resources: {:?}", execution_resources);
 
-    if stwo_proof {
-        // Create a performance tracing span for proof generation
-        let proof_span = tracing::span!(tracing::Level::INFO, "stwo_proof_generation");
-        let _proof_span_guard = proof_span.enter();
+    let prover_input_info =
+        cairo_runner.get_prover_input_info().expect("Unable to get prover input info");
 
-        // Convert CairoRunner to Stwo ProverInput
-        let cairo_input = stwo_cairo_adapter::plain::adapt_finished_runner(cairo_runner)
-            .map_err(|e| PyErr::new::<pyo3::exceptions::PyRuntimeError, _>(e.to_string()))?;
-
-        let prover_execution_resources = ProverExecutionResources::from_prover_input(&cairo_input);
-        tracing::debug!("Prover Execution resources: {:#?}", prover_execution_resources);
-
-        let ProverParameters { channel_hash: _, pcs_config, preprocessed_trace } =
-            default_prod_prover_parameters();
-
-        // Generate the proof
-        let proof =
-            prove_cairo::<Blake2sMerkleChannel>(cairo_input, pcs_config, preprocessed_trace)
-                .map_err(|e| PyErr::new::<pyo3::exceptions::PyRuntimeError, _>(e.to_string()))?;
-        drop(_proof_span_guard);
-        tracing::info!("Proof generation completed");
-
-        let proof_path = proof_path.unwrap_or_else(|| output_dir.join("proof.json"));
-        std::fs::write(
-            &proof_path,
-            serde_json::to_string(&proof)
-                .map_err(|e| PyErr::new::<pyo3::exceptions::PyRuntimeError, _>(e.to_string()))?,
-        )?;
-        tracing::info!("Proof written to {}", proof_path.display());
-
-        // Optional verification
-        if verify {
-            let verify_span = tracing::span!(tracing::Level::INFO, "proof_verification");
-            let _verify_span_guard = verify_span.enter();
-
-            verify_cairo::<Blake2sMerkleChannel>(proof, pcs_config, preprocessed_trace)
-                .map_err(|e| PyErr::new::<pyo3::exceptions::PyRuntimeError, _>(e.to_string()))?;
-            drop(_verify_span_guard);
-            tracing::info!("Proof verified successfully");
-        }
-        return Ok(());
-    }
-
-    // Create a performance tracing span for output writing
-    let output_span = tracing::span!(tracing::Level::INFO, "write_outputs");
-    let _output_span_guard = output_span.enter();
+    // Write prover input infos
+    std::fs::write(
+        output_dir.join("prover_input_infos.json"),
```

Comment:
will this always overwrite the last proof?

Reply:
fixd